teps/0050-ignore-task-failures.md

```diff
@@ -187,6 +194,222 @@ control over the task definitions but may desire to ignore a failure and continu
 
   ![Jenkins Dashboard](images/0050-jenkins-dashboard-with-failure-stage.png)
 
+## Proposal
+We propose a new filed ```OnFail``` to the [PipelineTask](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#adding-tasks-to-the-pipeline) definition.
+
+```go
+type PipelineTask struct {
+	Name string `json:"name,omitempty"`
+
+	//...
+
+	// OnFail defines the termination behavior of a pipeline when the task is failed
+	// can be set to [ continue | stopAndFail ]
+	OnFail OnFailType `json:onFail, "omitempty"`
+}
+```
+
+```go
+type OnFailType string
+
+const (
+	// StopAndFail indicates to stop the pipeline if the task is failed
+	StopAndFail OnFailType = "stopAndFail"
+	// Continue indicates to continue executing the rest of the pipeline irrespective of the status of the task
+	Continue    OnFailType = "continue"
+)
+```
+
+Pipeline author can set the ```OnFailType``` field to configure the task failure strategy. If set to ```StopAndFail```, the pipeline is stopped and failed when the task is failed. If set to ```Continue```, the failure of task is ignored and the pipeline continues to execute the rest of the DAG.
+
+```yaml
+- name: task1
+  OnFail: continue
+  taskSpec:
+    steps:
+      - image: alpine
+        name: exit-with-1
+        script: |
+          exit 1
+
+```
+This new field ```OnFail``` will be implemented as a alpha feature and can be enabled by setting enable-api-fields to alpha.
+
+Setting ```OnFail``` is optional, the default pipeline behaviour is ```StopAndFail```
+
+The task run information is avaialble under the ```pipelineRun.status.taskRuns```. Note that the original task run status remains as it is irrelevant of the value of ```OnFail``` (i.e. a failed task with ```OnFail: continue``` is still marked as failed). A prefix "[Task is failed but ignored]" will be added as a prefix to the original ```message``` of the task.
+
+ The task would be considered "successful" ONLY for the purposes of determining the status of the pipeline, which is represented in ```pipelineRun.status.conditions``` (if using ```full``` embedded status) or ```pipelineRun.status.childReferences``` (if using ```minimum``` embedded status). Details can be found in [TEP-0100](https://github.com/tektoncd/community/blob/main/teps/0100-embedded-taskruns-and-runs-status-in-pipelineruns.md).
+
+
+To distinguish pipeline run messages with and without ignored task failures, we explicitly add the ignored task failure count to ```pipelineRun.status.conditions.message``` in the following way if ignored task failure > 0: 
+
+```
+"Tasks Completed: A (Failed: B (C is ignored), Cancelled D), Skipped: E"
+```
```

Comment:
How would this look like when we have more than one tasks to ignore failure:

```
(C is ignored)
```

I suggest to change it to:

```
(Ignored: C)
```

Reply:
Oh, good catch! Will change it